server/src/core/link-generator/link-generator-service.ts

```diff
@@ -12,13 +16,21 @@ export class Links implements LinkGeneratorService {
     this.args = args;
   }
 
-  getUserDonateLink(user: User, amount: number): string {
+  async getUserDonateLink(user: User, amount: number): Promise<string> {
     const { name, email, phone } = user;
-    return this.getDonateLink({ name, email, phone, amount });
+    try {
+      const longLink = this.getDonateLink({ name, email, phone, amount });
+      const shortLink = await this.args.bitly.shortenLink(longLink);
```

Comment:
should be `this.args.shortner.shortenLink(longLink)`. It should not assume that it uses the bitly implementation. We could have testshortener that's used in unit tests that doesn't call an API.